### ISSUE TITLE
zmuser db credentials

### ISSUE THREAD

**Issue by veehexx** [QuantumObject/docker-zoneminder#65]
I recently migrated away from my own zoneminder VM into your docker image and found there doesn't appear to be a (documented) way to set the zmuser db credentials. I use non-default creds for the zmuser. Every time the container starts up, i have to re-apply the ZM_DB_PASS password which will then allow the WebUI to work correctly.

'-e MYSQL_PASSWORD=MyPassword' is defined at container start, and sql db is fine as the zm_create.sql script successfully ran. It's just the zm.conf file that doesn't use the correct password.

specifically in /etc/zm/zm.conf, the following:
```
# ZoneMinder database user
ZM_DB_USER=zmuser

# ZoneMinder database password
ZM_DB_PASS=zmpass
```

**Comment by angelrr7702** [comment]
at the moment I have this variable that can be modified by client : 

ZM_DB_HOST   ==> default value 'localhost'
MYSQL_ROOT_PASSWORD ==> default value 'mysqlpsswd'
MYSQL_ROOT  ==> default value 'root'

This is used like this without possibility to change at the moment 
     - MYSQL_USER=zmuser
     - MYSQL_PASSWORD=zmpass
     - MYSQL_DATABASE=zm

backup.sh ==> I need to change to make sure it work correctly when variable different that default .

**Comment by raulvasquez** [comment]
Agreed. The zm.conf should not be hardcoded for zmuser / zmpass and should pickup on container env variables set by the container. I have to access the containers command line and edit zm.conf to use my username and password currently.

**Comment by Dinth** [comment]
Im also waiting for this to be changed.

**Comment by tehniemer** [comment]
I'm running this with linuxserver/mariadb and keep getting the following error when starting zoneminder 

```
ERROR 1045 (28000): Access denied for user 'root'@'zoneminder.qnet-static' (using password: YES)
```

my environment variables are setup as follows

zoneminderdb
```
MYSQL_ROOT_PASSWORD = *********
MYSQL_DATABASE = zmdb
MYSQL_USER = zoneminder
MYSQL_PASSWORD = **********
MYSQL_ROOT_HOST = %
```

zoneminder
```
ZM_DB_HOST = zoneminderdb
```

I'm thinking this is somewhat related as I have three other containers running off separate mariadb instances without issue.

**Comment by moogoos** [comment]
I just came across this and is a huge dealbreaker for me on this image...... bummer good work otherwise but hardcoded creds? And why the need for root password if we pregen an external db...

**Comment by angelrr7702** [comment]
branch 1.34-db-fix will be used to fix this problem and other , still no available at docker hub but it is used to be build local for testing .. , when done, it will be merge with master and 1.34.

**Comment by angelrr7702** [comment]
quantumobject/docker-zoneminder:1.34-db-fix   ==> available for testing and troubleshooting ....

allow to change the dafault value of :
 
ENV ZM_DB_HOST db

ENV ZM_DB_NAME zm 

ENV ZM_DB_USER zmuser

ENV ZM_DB_PASS zmpass

with -e at docker run  command ...

**Comment by angelrr7702** [comment]
move 1.34-db-fix to master , 3.42 and 3.44 and docker hub .
